static/app/views/dashboardsV2/widgetCard/widgetQueries.tsx

```diff
@@ -115,31 +115,37 @@ function WidgetQueries({
   const config = ErrorsAndTransactionsConfig;
   const context = useContext(DashboardsMEPContext);
 
-  let isMetricsData: undefined | boolean;
   let setIsMetricsData: undefined | ((value?: boolean) => void);
 
   if (context) {
-    isMetricsData = context.isMetricsData;
     setIsMetricsData = context.setIsMetricsData;
   }
 
-  const afterFetchSeriesData = useCallback(
-    (rawResults: SeriesResult) => {
-      // If one of the queries is sampled, then mark the whole thing as sampled
-      const currentResultIsMetricsData =
-        isMetricsData === false ? false : getIsMetricsDataFromSeriesResponse(rawResults);
-      setIsMetricsData?.(currentResultIsMetricsData);
-    },
-    [isMetricsData, setIsMetricsData]
-  );
+  const isSeriesMetricsDataResults: boolean[] = [];
+  const afterFetchSeriesData = (rawResults: SeriesResult) => {
+    if (rawResults.data) {
+      if (rawResults.isMetricsData !== undefined) {
+        isSeriesMetricsDataResults.push((rawResults as EventsStats).isMetricsData!);
+      }
+    } else {
+      Object.keys(rawResults).forEach(key => {
+        const rawResult: EventsStats = rawResults[key];
+        if (rawResult.isMetricsData !== undefined) {
+          isSeriesMetricsDataResults.push(rawResult.isMetricsData!);
```

Comment:
```suggestion
          isSeriesMetricsDataResults.push(rawResult.isMetricsData);
```
We don't need `!` here because line 132 tells TS it's an EventsStats type